scripts/docs/build-dev-center-docs.ts

```diff
@@ -0,0 +1,112 @@
+import * as _ from "lodash";
+import * as path from "path";
+import * as fs from "fs-extra";
+import { Application, TSConfigReader, TypeDocOptions } from "typedoc";
+
+// Options added by the Typedoc Markdown Plugin
+interface MarkdownPluginOptions {
+    skipSidebar: boolean;
+    hideBreadcrumbs: boolean;
+}
+
+interface DevCenterDocsOptions {
+    version: string;
+    bundleHash?: string;
+    options?: Partial<TypeDocOptions>;
+    flagDocsAsPreview: boolean;
+    outputDir: string;
+}
+
+export const buildDevCenterDocs = async ({
+    version,
+    bundleHash,
+    flagDocsAsPreview,
+    outputDir,
+    options
+}: DevCenterDocsOptions) => {
+    const app = new Application();
+
+    // So TypeDoc can load tsconfig.json
+    app.options.addReader(new TSConfigReader());
+
+    const docsConfig: Partial<TypeDocOptions & MarkdownPluginOptions> = {
+        ...options,
+        theme: "docusaurus", // Theme that generates output closest to Dev Center format
+        skipSidebar: true, // Disable sidebar generation
+        hideBreadcrumbs: true, // Dev Center handles breadcrumbs for us
+        readme: "none",
+        plugin: [
+            'typedoc-plugin-markdown',
+            'typedoc-plugin-no-inherit'
+        ],
+    };
+
+    app.bootstrap(docsConfig);
+
+    const project = app.convert(app.expandInputFiles(["src"]));
+    if (!project) throw new Error("Unable to generate typedoc project");
+
+    // Rendered docs
+    app.generateDocs(project, outputDir);
+
+    // Flatten typedoc's generated classes folder
+    const classesDir = path.join(outputDir, "classes");
+    const fileNames = await fs.readdir(classesDir);
+    await Promise.all(
+        fileNames.map(async (name) =>
+            fs.rename(path.join(classesDir, name), path.join(outputDir, name))
+        )
+    );
+
+    // Delete typedoc's generated classes folder
+    await fs.remove(classesDir);
+
+    // Use custom index file in place of typedoc's index
+    await fs.copy("doc/index.md", path.join(outputDir, "index.md"));
+
+    // Add header to index file
+    await transformFiles(outputDir, buffer => "---\ntitle: Client App SDK\n---\n\n" + buffer);
```

Comment:
Looks like we're actually adding the title to every doc file.  Any issue with this with devCenter?  If so, we'd need something else here so prepend will work.  It assumes a leading `---`.

Reply:
Looks like the generator creates some metadata automatically.  So, if we add this initial block to all files unconditionally, looks like we create multiple metadata blocks.

Reply:
Whooops, not sure how that happened. This was supposed to only have been run on the index file. But I def now see that it's running on everything 😅

Reply:
Should be g2g here now 👍  @justinaray